embedded-graphics/MIGRATING-0.5-0.6.md

```diff
@@ -0,0 +1,248 @@
+- A new `size()` method is now a required item. This should return the width and height of the display as a `Size`.
+- The implementation of `DrawTarget` must now provide an associated `Error` type. In the above example `core::convert::Infallible` is used, however a better error type should be used to communicate hardware failures, etc to the user.
+- Any pixels that are offscreen (negative coordinates or greater than display dimensions) should result in a noop. In the example above, `self.set_pixel()` (defined by the SSD1306 driver) will not attempt to set any pixels beyond the positive screen limits.
+- `draw_pixel()` now returns `Result<(), Self::Error>` to account for driver error handling.
+
+## For font authors
+
+- The `embedded_graphics::fonts::font_builder` module along with its exported `FontBuilderConf` and `FontBuilder` is removed. Now only `Font` needs to be implemented.
+- The `Font` trait (used to be `FontBuilderConf`) implementation has changed slightly, replacing `CHART_WIDTH` and `CHAR_HEIGHT` with a single `CHARACTER_SIZE` constant, using the `Size` struct:
+
+  ```diff
+  - use embedded_graphics::fonts::font_builder::{FontBuilder, FontBuilderConf};
+  + use embedded_graphics::fonts::Font;
+
+  + use embedded_graphics::geometry::Size;
+
+    #[derive(Debug, Copy, Clone)]
+    pub enum MyFont {}
+
+    impl Font for MyFont {
+        const FONT_IMAGE: &'static [u8] = include_bytes!("../data/my_font.raw");
+  -     const CHAR_WIDTH: u32 = 12;
+  -     const CHAR_HEIGHT: u32 = 22;
+  +     const CHARACTER_SIZE: Size = Size::new(12, 22);
+        const FONT_IMAGE_WIDTH: u32 = 480;
+    }
+  ```
+
+## For library consumers
+
+### General drawing operations
+
+Drawing operations are "reversed" in 0.6.0. Instead of calling `display.draw(thing)`, call `thing.draw(&mut display)`:
+
+```diff
+- let rect = Rectangle::new(Coord::new(50, 20), Coord::new(60, 35))
+-     .stroke(Some(5u8))
+-     .stroke_width(3)
+-     .fill(Some(10u8));
+-
+- display.draw(rect);
+
++ let style = PrimitiveStyleBuilder::new()
++     .stroke_color(Rgb565::RED)
++     .stroke_width(3)
++     .fill_color(Rgb565::GREEN)
++     .build();
++
++ Rectangle::new(Point::new(50, 20), Point::new(60, 35))
++     .into_styled(style)
++     .draw(&mut display)?;
+```
+
+Drawing operations are now fallible, with `.draw()` calls returning a `Result`. This allows for error handling if an error occurs during a drawing operation. The error type of this `Result` is dependent on the associated `Error` type as defined by the display driver.
+
+### Coordinates and positioning
+
+The `Coord` and `UnsignedCoord` have been renamed to `Point` and `Size` respectively. Both items are no longer tuple structs, but structs with the named fields `x` and `y` for `Point` and `width` and `height` for `Size`. `Point`s can store negative coordinates, whereas `Size`s must be positive up to `u32::MAX_VALUE`.
```

Comment:
IMO they weren't just renamed but replaced by the new types and gained some semantic meaning.

An important aspect that is missing here is the change to the `Pixel` type. It used `UnsignedCoord` in 0.5 and now uses the signed `Point` instead.